src/client/pkg/grpcutil/addr.go

```diff
@@ -0,0 +1,127 @@
+package grpcutil
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+const (
+	// DefaultPachdNodePort is the pachd kubernetes service's default
+	// NodePort.Port setting
+	DefaultPachdNodePort = 30650
+
+	// DefaultPachdPort is the pachd kubernetes service's default
+	// Port (often used with Pachyderm ELBs)
+	DefaultPachdPort = 650
+)
+
+var (
+	// ErrNoPachdAddress is returned by ParsePachdAddress when the input is an
+	// empty string
+	ErrNoPachdAddress = errors.New("no pachd address specified")
+	// DefaultPachdAddress is the default PachdAddress that should be used
+	// if none is otherwise specified. It's a loopback that should rely on
+	// port forwarding.
+	DefaultPachdAddress = PachdAddress{
+		Secured: false,
+		Host:    "0.0.0.0",
+		Port:    DefaultPachdNodePort,
+	}
+)
+
+// PachdAddress represents a parsed pachd address value
+type PachdAddress struct {
+	// Secured specifies whether grpcs should be used
+	Secured bool
+	// Host specifies the pachd address host without the port
+	Host string
+	// Port specifies the pachd port
+	Port uint16
+}
+
+// ParsePachdAddress parses a string into a pachd address, or returns an error
+// if it's invalid
+func ParsePachdAddress(value string) (PachdAddress, error) {
+	if value == "" {
+		return PachdAddress{}, ErrNoPachdAddress
+	}
+
+	secured := false
+
+	if strings.Contains(value, "://") {
+		// only parse the url if it contains a scheme, as net/url doesn't
+		// appropriately handle values without one
+
+		u, err := url.Parse(value)
+		if err != nil {
+			return PachdAddress{}, fmt.Errorf("could not parse pachd address: %v", err)
+		}
+
+		if u.Scheme != "grpc" && u.Scheme != "grpcs" {
+			return PachdAddress{}, fmt.Errorf("unrecognized scheme in pachd address: %s", u.Scheme)
```

Comment:
Do you think it would make sense to allow `http` and `https`? I think technically GRPC is layered on HTTP2. Also, if we don't allow `http`, perhaps the error should specify that the scheme must be `grpc` or `grpcs` (I can just see somebody using `http://whatever` and being like "what do you mean unrecognized scheme?")